components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIDefinitionFromSwagger20.java

```diff
@@ -208,6 +210,21 @@ private void populateConfigMapForScopes(Swagger swagger, String namespace) {
     }
 
     @Override
+    public Map<String, String> getScope(String resourceConfigJSON) throws APIManagementException {
+        SwaggerParser swaggerParser = new SwaggerParser();
+        Swagger swagger = swaggerParser.parse(resourceConfigJSON);
+        String basePath = swagger.getBasePath();
+        String nameSpace = getNamespaceFromBasePath(basePath);
+        if (nameSpace == null) {
+            return new HashMap<>();
+        }
+        Map<String, SecuritySchemeDefinition> securityDefinitions = swagger.getSecurityDefinitions();
+        Map.Entry<String, SecuritySchemeDefinition> entry = securityDefinitions.entrySet().iterator().next();
```

Comment:
can this lead to a possoble null pointer if the secrity definitions not present?